e2e/nomostest/new.go

```diff
@@ -318,8 +316,7 @@ func FreshTestEnv(t nomostesting.NTB, opts *ntopts.New) *NT {
 	if err := nt.KubeClient.Create(fake.NamespaceObject(metrics.MonitoringNamespace)); err != nil {
 		nt.T.Fatal(err)
 	}
-
-	if nt.GitProvider.Type() == e2e.Local {
+	if *e2e.GitProvider == e2e.Local {
```

Comment:
Not for this PR, but it might be worth thinking if this logic to set up the auth for git can also be abstracted.

Reply:
Agreed, that instantiation could be abstracted to the provider as well